### ISSUE TITLE
insecureKubeletReadonlyPortEnabled support

### ISSUE THREAD

**Issue by wyardley** [terraform-google-modules/terraform-google-kubernetes-engine#2013]
### TL;DR

Is the `insecureKubeletReadonlyPortEnabled` flag in the kubelet config supported? With the recent announcements
https://cloud.google.com/kubernetes-engine/docs/how-to/disable-kubelet-readonly-port#migrate-apps, I updated the cluster configs.

Probably depends on https://github.com/hashicorp/terraform-provider-google/issues/15208

After running:
```
% gcloud container clusters update xxx \
    --location=us-central1 --project xxx \
    --no-enable-insecure-kubelet-readonly-port
% gcloud container node-pools update primary \     
    --cluster=cluster --project xxx \
    --location=us-central1 \                  
    --no-enable-insecure-kubelet-readonly-port
```

I seem to be getting some unresolvable diffs with the kubelet config. I might be just running into something related to https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1922 but I believe I did check the configs after updating to the latest module version, and I didn't have any luck just by setting `pod_pids_limit` and `cpu_cfs_quota` in the pod config.

diffs:
```
          - kubelet_config {
              - cpu_cfs_quota  = false -> null
              - pod_pids_limit = 0 -> null
            }
```

I can see this in the debug logs
```
2024-07-26T11:40:22.622-0700 [DEBUG] provider.terraform-provider-google:   "kubeletConfig": {
2024-07-26T11:40:22.622-0700 [DEBUG] provider.terraform-provider-google:    "insecureKubeletReadonlyPortEnabled": false
2024-07-26T11:40:22.622-0700 [DEBUG] provider.terraform-provider-google:   },
```

**Comment by DrFaust92** [comment]
seems this depends on open PR https://github.com/GoogleCloudPlatform/magic-modules/pull/11272 so its not an issue on the module

**Comment by ebeltramo96** [comment]
we had to revert the configuration for each nodepool because we were not able to apply the TF plans anymore:

`gcloud container node-pools update nodepool-1 \
     --cluster=cluster-1 --location=location-1 \
     --enable-insecure-kubelet-readonly-port`

looking forward to the TF implementation

**Comment by wyardley** [comment]
Side note: with latest provider and module, I was getting a permadiff closer to this (when that setting was changed); presumably because another field under `nodeConfig.kubeletConfig` is returning something)

```diff
      ~ node_config {
         [...]
          ~ kubelet_config {
              + cpu_manager_policy = "static"
                # (2 unchanged attributes hidden)
            }

            # (2 unchanged blocks hidden)
        }
```

For me, setting that to the undocumented value of `""` in the first node pool solved the permadiff (and doesn't seem to cause issues with other settings, including the insecure kubelet one) in the short term.

Also filed a separate issue for that [here](https://github.com/hashicorp/terraform-provider-google/issues/19225)

**Comment by wyardley** [comment]
@DrFaust92 Should be supported in upcoming 6.x releases and is also supposed to get cherry-picked into 5.x

(also, sorry for causing some conflicts for your GCFS PR)

**Comment by wyardley** [comment]
It will be worth keeping an eye out for issues (in the module) related to https://github.com/hashicorp/terraform-provider-google/issues/15767 once this is supported.

**Comment by wyardley** [comment]
FWIW, this is now out in `v5.44.0` and `v6.2.0`.

https://github.com/hashicorp/terraform-provider-google/releases/tag/v5.44.0
https://github.com/hashicorp/terraform-provider-google/releases/tag/v6.2.0

I'm throwing up a pass at adding the support

That said, there may be some quirks when people start adding the setting, especially before
https://github.com/hashicorp/terraform-provider-google/issues/15767
https://github.com/hashicorp/terraform-provider-google/issues/19225
are resolved -- setting `cpu_manager_policy` explicitly to `""` when it's not set in the module might help as a short term workaround, though (see comment again).

**Comment by UDtorrey** [comment]
similar to OP, trying to disable the insecure kubelet readonly port.

looking at [feat(TPG>=5.44)!: add support for insecureKubeletReadonlyPortEnabled](https://github.com/wyardley/terraform-google-kubernetes-engine/commit/2071fc9259a3ae921e5d9717753a040d1e17f141),  I see this was updated in numerous modules, `beta-private-cluster`, `beta-public-cluster`, `private-cluster-update-variant`, etc.

We use [beta-autopilot-private-cluster](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/tree/v33.0.3/modules/beta-autopilot-private-cluster), but i dont see that updated? Perhaps I'm missing it, or it's downstream of `beta-autopilot-private-cluster`

Is there guidance on how to proceed with modules that were not updated?
Will `insecure_kubelet_readonly_port_enabled` be supported in all modules, or just a select few? 

Note: I have not executed the `gcloud container clusters update xxx --no-enable-insecure-kubelet-readonly-port` to identify any resulting drift from the TF.  I'd prefer not to create drift (due to internal politics).
